### PR TITLE
Convert to .md: draft_export.rst, export.ir_spec.rst, fft.rst

### LINKED ISSUE #155020: Convert to markdown: draft_export.rst, export.ir_spec.rst, export.programming_model.rst, export.rst, fft.rst
Convert the following files to MyST markdown and ensure that the doc tests are green:

- [draft_export.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/draft_export.rst)
- [export.ir_spec.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/export.ir_spec.rst)
- [export.programming_model.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/export.programming_model.rst)
- [export.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/export.rst)
- [fft.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/fft.rst)

To convert an `.rst` file:

1) Assign the issue to yourself using the `/assigntome` comment.

2) **Important!** Use `git mv filename.rst filename.md` to convert the issue from `.rst` to `.md`. **DO NOT delete**
   the `.rst` and create the `.md` - we won't accept PRs with deleted `.rst` files. We want to preserve the git log of
   those files. Therefore, using `git mv` is very important!

3) Before submitting a PR, build the docs locally as described in the [README](https://github.com/pytorch/pytorch/blob/main/README.md#building-the-documentation).
   Preview localy with `make serve` to ensure that the HTML renders correctly.    

4) Run `lintrunner` from the root of the repository:

   ```
   pip install lintrunner
   lintrunner init
   lintrunner -m main
   ```

4) Submit a PR with your changes and ensure the PR passes the doc CI tests. Specifically it needs to pass:

   * pull / linux-docs / build-docs-python-false (pull_request)
   * linux-jammy-py3.9-gcc11 / test (docs_test, 1, 1, lf.linux.2xlarge)

   Doc tests start after the Docker build, which might take about 30 minutes. 
   The doc build itself should take about 25 minutes. When the build is done, 
   you can see the preview of your build by clicking:
   📄 **Preview Python docs built from this PR** in the `pytorch-bot` comment.

   Make sure your PR is tagged with `docathon-h1-2025`, `module: docs`, `topic: not user facing`,
   and `topic: docs` labels.

5) Compare the preview of the pages you converted with same pages it the main branch.
   The converted pages in preview must look the same as in the main branch.

6) Once the PR is accepted, one of the maintainers will merge it.

See [Quick MyST Conversion Reference](https://github.com/pytorch/pytorch/wiki/Docstring-Guidelines#restructuredtext-vs-markdown)
in our wiki or [MyST Parser documentation](https://myst-parser.readthedocs.io/en/v0.15.1/sphinx/use.html) for detailed syntax.

Here is an example [PR](https://github.com/pytorch/pytorch/pull/154438)

### PR DESCRIPTION
**Pull request by j-silv**
Used [rst2myst tool](https://rst-to-myst.readthedocs.io/en/latest/)

Fixes #155020. This PR is split into 3 to pass sanity check.

Docs comparison (check out the 'new' whenever docs build)

1. draft_export ([old](https://docs.pytorch.org/docs/main/draft_export.html) vs. [new](https://docs-preview.pytorch.org/pytorch/pytorch/155567/draft_export.html)) 
2. export.ir_spec ([old](https://docs.pytorch.org/docs/main/export.ir_spec.html) vs. [new](https://docs-preview.pytorch.org/pytorch/pytorch/155567/export.ir_spec.html))
3. fft ([old](https://docs.pytorch.org/docs/main/fft.html) vs. [new](https://docs-preview.pytorch.org/pytorch/pytorch/155567/fft.html))

cc @svekars @sekyondaMeta @AlannaBurke

### CODE REVIEW

File: docs/source/export.ir_spec.md [hunk 1]
```diff
@@ -44,154 +41,155 @@ This documentation will primarily focus on highlighting areas where Export IR
...
 This format captures everything present in the Node class, with the exception of
-``meta``, in a compact format.
+`meta`, in a compact format.
 
 Concretely:
 
-- **<name>** is the name of the node as it would appear in ``node.name``.
-
-- **<op_name>** is the ``node.op`` field, which must be one of these:
+- **\<name>** is the name of the node as it would appear in `node.name`.
```
Comment (svekars) [inline docs/source/export.ir_spec.md]:
```suggestion
- **<name>** is the name of the node as it would appear in `node.name`.
```

File: docs/source/export.ir_spec.md [hunk 2]
```diff
@@ -44,154 +41,155 @@ This documentation will primarily focus on highlighting areas where Export IR
...
+- **\<op_name>** is the `node.op` field, which must be one of these:
```
Comment (svekars) [inline docs/source/export.ir_spec.md]:
```suggestion
- **<op_name>** is the `node.op` field, which must be one of these:
```

File: docs/source/export.ir_spec.md [hunk 3]
```diff
@@ -44,154 +41,155 @@ This documentation will primarily focus on highlighting areas where Export IR
...
   `<call_function>`, `<placeholder>`,
   `<get_attr>`, or `<output>`.
-
-- **<target>** is the target of the node as ``node.target``. The meaning of this
-  field depends on ``op_name``.
-
-- **args1, … args 4…** are what is listed in the ``node.args`` tuple. If a
-  value in the list is an :class:`torch.fx.Node`, then it will be especially
+- **\<target>** is the target of the node as `node.target`. The meaning of this
```
Comment (svekars) [inline docs/source/export.ir_spec.md]:
```suggestion
- **<target>** is the target of the node as `node.target`. The meaning of this
```

File: docs/source/export.md [hunk 4]
```diff
@@ -0,0 +1,923 @@
...
+  computational graph.
+- **Graph properties**: The graph is purely functional, meaning it does not
+  contain operations with side effects such as mutations or aliasing. It does
+  not mutate any intermediate values, parameters, or buffers.
+- **Metadata**: The graph contains metadata captured during tracing, such as a
+  stacktrace from user's code.
+
+Under the hood, `torch.export` leverages the following latest technologies:
+
+- **TorchDynamo (torch.\_dynamo)** is an internal API that uses a CPython feature
```
Comment (svekars) [inline docs/source/export.md]:
```suggestion
- **TorchDynamo (torch._dynamo)** is an internal API that uses a CPython feature
```

**Review by svekars — COMMENTED** [review]
This looks pretty good. `pr-sanity check` is failing - need to move one file out of this pr, say export-programming_model.

File: docs/source/export.ir_spec.md [hunk 5]
```diff
@@ -1,7 +1,6 @@
-.. _export.ir_spec:
+(export-ir-spec)=
```
Comment (svekars) [inline docs/source/export.ir_spec.md]:
```suggestion
(export.ir-spec)=
```

File: docs/source/export.md [hunk 6]
```diff
@@ -0,0 +1,923 @@
...
+
+export(M(), (torch.ones(3, 3),), strict=False)  # Non-strict traces successfully
+export(M(), (torch.ones(3, 3),))  # Strict mode fails with torch._dynamo.exc.Unsupported: ContextManager
+```
+
+In this example, the first call using non-strict mode (through the
+`strict=False` flag) traces successfully whereas the second call using strict
+mode (default) results with a failure, where TorchDynamo is unable to support
+context managers. One option is to rewrite the code (see {ref}`Limitations of torch.export <Limitations of
+torch.export>`), but seeing as the context manager does not affect the tensor
```
Comment (svekars) [inline docs/source/export.md]:
```suggestion
context managers. One option is to rewrite the code (see {ref}`Limitations of torch.export <limitations-of-torch-export>`), but seeing as the context manager does not affect the tensor
```

File: docs/source/export.md [hunk 7]
```diff
@@ -0,0 +1,923 @@
...
+```
+
+The conditional of (`x.shape[0] > 5`) does not appear in the
+`ExportedProgram` because the example inputs have the static
+shape of (10, 2). Since `torch.export` specializes on the inputs' static
+shapes, the else branch (`x - 1`) will never be reached. To preserve the dynamic
+branching behavior based on the shape of a tensor in the traced graph,
+{func}`torch.export.Dim` will need to be used to specify the dimension
+of the input tensor (`x.shape[0]`) to be dynamic, and the source code will
+need to be {ref}`rewritten <Data/Shape-Dependent Control Flow>`.
```
Comment (svekars) [inline docs/source/export.md]:
```suggestion
need to be {ref}`rewritten <data-shape-dependent-control-flow>`.
```

File: docs/source/export.md [hunk 8]
```diff
@@ -0,0 +1,923 @@
...
+tracing is based on TorchDynamo, which evaluates at the Python
+bytecode level, there will be significantly fewer rewrites required compared to
+previous tracing frameworks.
+
+When a graph break is encountered, {ref}`ExportDB <torch.export_db>` is a great
+resource for learning about the kinds of programs that are supported and
+unsupported, along with ways to rewrite programs to make them traceable.
+
+An option to get past dealing with this graph breaks is by using
+{ref}`non-strict export <Non-Strict Export>`
```
Comment (svekars) [inline docs/source/export.md]:
```suggestion
{ref}`non-strict export <non-strict-export>`
```

File: docs/source/export.md [hunk 9]
```diff
@@ -0,0 +1,923 @@
+(torch-export)=
```
Comment (svekars) [inline docs/source/export.md]:
```suggestion
(torch.export)=
```

**Comment by j-silv** [comment]
Sorry about missing all the small fixes @svekars. The rst2myst tool is not perfect and I missed those from visual inspection. I'm having some issues running the tests locally, so that's also leading to a lot of these little typos/broken links.

Also note, I screwed up linking to the new docs preview in pretty much all my PRs (copy and paste mix-up). I was linking to the wrong pull request ID. Most of the docs never got built though so I don't know if this is a big deal. I'm fixing the links now.

**Comment by svekars** [comment]
No worries, you are doing great - I actually thought that rst2myst tool is doing a great job overall. pr-sanity-check is still failing though, probably need to split to yet enough PR with export.rst

**Review by svekars — APPROVED** [review]
These three fils LGTM!
